messages/src/lib.rs

```diff
@@ -1520,23 +1363,22 @@ impl<Q: QueryType> CollectionReq<Q> {
     }
 }
 
-impl<Q: QueryType> Encode for CollectionReq<Q> {
+impl<Q: QueryType> Encode for CollectReq<Q> {
     fn encode(&self, bytes: &mut Vec<u8>) {
+        self.task_id.encode(bytes);
         self.query.encode(bytes);
         encode_u32_items(bytes, &(), &self.aggregation_parameter);
```

Comment:
The aggregation parameter's length prefix needs to change from 4 bytes to 2 bytes here, in `AggregateShareReq`, and in `CollectReq`.

Reply:
Excellent attention to detail, thanks -- I knew I was going to miss something like this.

I also updated the lengths in `AggregateInitializeReq`.